src/default-config-backend.cpp

```diff
@@ -58,18 +59,22 @@ static int handle_config_updated(int fd, uint32_t mask, void *data)
          ptr += sizeof(inotify_event) + event->len)
     {
         event = reinterpret_cast<inotify_event*>(ptr);
-        // We reload in two main cases:
+        // We reload in two cases:
         //
-        // - Config file itself was modified
-        // - Config file was created inside parent directory
-        should_reload |=
-            (event->wd == wd_cfg_file) || (cfg_file_basename == event->name);
+        // 1. The config file itself was modified, or...
+        should_reload |= event->wd == wd_cfg_file;
+        // 2. The config file was moved/created inside the parent directory.
+        if (event->len > 0) {
+            // This is UB unless event->len > 0.
+            auto name_matches = strcmp(cfg_file_basename.c_str(), event->name);
```

Comment:
Wait, strcmp returns 0 iff the config file and the name match, isn't that right? So, the variable is actually `name_does_not_match`

Reply:
....and that's what was going on. `==` never succeeded whenever it was supposed to. I'll revise the patch to make it actually correct. Thanks!